Google.PowerShell/Common/ReportCmdletResults.cs

```diff
@@ -0,0 +1,189 @@
+        {
+            var expectedRecord = EventRecord.Create(cmdletName, parameterSet, errorCode);
+            for (int i = 0; i < _events.Length; i++)
+            {
+                if (_events[i].Equals(expectedRecord))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+    }
+
+    /// <summary>
+    /// Type of analytic event.
+    /// </summary>
+    public enum AnalyticsEventCategory
+    {
+        CmdletInvocation,
+        Provider
+    }
+
+    /// <summary>
+    /// Reports PowerShell cmdlet results to Google Analytics.
+    /// </summary>
+    public class GoogleAnalyticsCmdletReporter : IReportCmdletResults
+    {
+        // Google Analytics Property ID, see di-metrics.
+        private const string PropertyId = "UA-36037335-1";
+        // Application name to report to GA.
+        private const string ApplicationName = "google-cloud-powershell";
+        // Version of the Google.PowerShell assembly.
+        private static readonly Lazy<string> s_appVersion = new Lazy<string>(() => Assembly.GetExecutingAssembly().GetName().Version.ToString());
+
+        /// <summary>
+        /// Analytics reporter object.
+        /// </summary>
+        private Lazy<AnalyticsReporter> _reporter = null;
+
+        /// <summary>
+        /// Category for all analytic events reported through this instance.
+        /// </summary>
+        private AnalyticsEventCategory _category = AnalyticsEventCategory.CmdletInvocation;
+
+        public GoogleAnalyticsCmdletReporter(string clientID, AnalyticsEventCategory analyticsCategory)
+        {
+            // We report results to Google Analytics if the user has opted-into Cloud SDK metric reporting. However, we
+            // don't want to report data during development. (e.g. when running unit tests.)
+            string envVar = Environment.GetEnvironmentVariable("DISABLE_POWERSHELL_ANALYTICS");
+            bool debugFlag = (envVar == "TRUE");
+
+            _category = analyticsCategory;
+            _reporter = new Lazy<AnalyticsReporter>(() =>
+            {
+                return new AnalyticsReporter(PropertyId,
+                    clientId: clientID,
+                    appName: ApplicationName,
+                    appVersion: s_appVersion.Value,
+                    debug: debugFlag);
+            });
+        }
+
+        public void ReportSuccess(string cmdletName, string parameterSet)
+        {
+            Report(_category, cmdletName, parameterSet, null);
+        }
+
+        public void ReportFailure(string cmdletName, string parameterSet, int errorCode)
+        {
+            Report(_category, cmdletName, parameterSet, errorCode);
+        }
+
+        private void Report(AnalyticsEventCategory category, string cmdletName, string parameterSet, int? errorCode)
+        {
+            _reporter.Value?.ReportEvent(
```

Comment:
You are supposed to use the `ReportPageView()` method instead, analytics are reported as virtual page views with more information than events can handle.